### PR TITLE
Aggies Sprint 9

### PR DESCRIPTION
**Pull request by wwelling**
[MODCAMUNDA-16](https://folio-org.atlassian.net/browse/MODCAMUNDA-16)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ mvn clean install
 
 cd mod-camunda
 docker build -t folio/mod-camunda .
-docker run -d -p 9000:9000 folio/mod-camunda
+docker run -d -p 9000:8081 folio/mod-camunda
```
Comment (kaladay) [inline README.md:30]:
```suggestion
docker run -d -p 9000:9000 folio/mod-camunda
```

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -2,6 +2,22 @@
   "id": "@project.artifactId@-@project.version@",
   "name": "Camnuda BPM Module",
   "provides": [
+    {
+      "id": "camunda-history",
```
Comment (kaladay) [inline descriptors/ModuleDescriptor-template.json:6]: These additions do not appear to be related to the referenced community sprint issue [MODCAMUNDA-16](https://folio-org.atlassian.net/browse/MODCAMUNDA-16).

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -2,6 +2,22 @@
...
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["GET"],
+          "pathPattern": "/camunda/history/process-instance",
+          "permissionsRequired": ["camunda.history.process-instance.collection.get"]
```
Comment (kaladay) [inline descriptors/ModuleDescriptor-template.json:12]: These permissions are created without consideration to ongoing plans:
- https://folio-org.atlassian.net/browse/MODCAMUNDA-4
- https://folio-org.atlassian.net/browse/MODWRKFLOW-15

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -348,28 +384,30 @@
...
-      { "name": "EVENT_QUEUE_NAME", "value": "event.queue", "description": "The ActiveMQ event queue name." },
+      { "name": "DB_HOST", "value": "postgres", "description": "Postgres hostname" },
+      { "name": "DB_PORT", "value": "5432", "description": "Postgres port" },
+      { "name": "DB_USERNAME", "value": "folio_admin", "description": "Postgres username" },
+      { "name": "DB_PASSWORD", "value": "folio_admin", "description": "Postgres username password" },
+      { "name": "DB_DATABASE", "value": "okapi_modules", "description": "Postgres database name" },
+      { "name": "DB_QUERYTIMEOUT", "value": "60000", "description": "Database query timeout." },
+      { "name": "DB_CHARSET", "value": "UTF-8", "description": "Database charset." },
+      { "name": "DB_MAXPOOLSIZE", "value": "16", "description": "Database max pool size." },
+      { "name": "KAFKA_HOST", "value": "kafka", "description": "Kafka broker hostname" },
```
Comment (kaladay) [inline descriptors/ModuleDescriptor-template.json:401]: This should described the `ENV` environment variable.

File: src/main/java/org/folio/rest/camunda/kafka/EventConsumer.java [hunk 5]
```diff
@@ -1,36 +1,39 @@
...
   @Autowired
   private RuntimeService runtimeService;
 
   @Autowired
   private ObjectMapper objectMapper;
 
-  @JmsListener(destination = "${event.queue.name}")
+  @KafkaListener(
+    id = "mod-camunda-events-listener",
+    containerFactory = "kafkaListenerContainerFactory",
```
Comment (kaladay) [inline src/main/java/org/folio/rest/camunda/kafka/EventConsumer.java:30]: Everything else is an environment variable.
Should the `containerFactory` be one?
Should it specifically not be an environment variable configurable setting?
Is there only a single factory to select from?

File: src/main/java/org/folio/rest/camunda/service/CamundaApiService.java [hunk 6]
```diff
@@ -54,6 +54,8 @@ public Workflow deployWorkflow(Workflow workflow, String tenant) throws Workflow
         Bpmn.writeModelToStream(System.out, modelInstance);
       }
 
+      logger.error("Failed to deploy workflow {}:{}. {}", workflow.getId(), workflow.getName(), e.getMessage(), e);
```
Comment (kaladay) [inline src/main/java/org/folio/rest/camunda/service/CamundaApiService.java:57]: Why is there an additional logger when the exception is already being thrown?

File: src/main/resources/application.yaml [hunk 7]
```diff
@@ -101,7 +103,8 @@ spring:
     mode: TEXT
     suffix: .sql
 
-  jersey.application-path: rest
+  # this must be `camunda` to match module descriptor path patterns
+  jersey.application-path: camunda
```
Comment (kaladay) [inline src/main/resources/application.yaml:107]: This will eventually need to become `workflow-engine` as per:
- https://folio-org.atlassian.net/browse/MODWRKFLOW-15
- https://folio-org.atlassian.net/browse/MODCAMUNDA-4

This is something for some future sprint to be aware of.

**Review by kaladay — CHANGES_REQUESTED** [review]

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (wwelling) [reply descriptors/ModuleDescriptor-template.json:6]: @jeremythuff can we get an issue for exposing these endpoints?

File: descriptors/ModuleDescriptor-template.json (hunk 4 above)
Comment (wwelling) [reply descriptors/ModuleDescriptor-template.json:401]: What should its value be? Why not use the application default?

File: src/main/java/org/folio/rest/camunda/kafka/EventConsumer.java (hunk 5 above)
Comment (wwelling) [reply src/main/java/org/folio/rest/camunda/kafka/EventConsumer.java:30]: No. It is statically defined as in the change in spring-module-core, https://github.com/folio-org/spring-module-core/pull/68/files#diff-a1decc886717a788edfcdd290a58393320070338247431439e6e50000dccee82R27.

File: src/main/java/org/folio/rest/camunda/service/CamundaApiService.java (hunk 6 above)
Comment (wwelling) [reply src/main/java/org/folio/rest/camunda/service/CamundaApiService.java:57]: Because not all the possible exceptions have an exception handler. 

`NotFoundException | NotValidException | ParseException | AuthorizationException e`

https://github.com/folio-org/mod-camunda/blob/master/src/main/java/org/folio/rest/camunda/controller/advice/WorkflowControllerAdvice.java#L17

This is a reasonable error log to add instead of updating controller advice which needs more planning on exception handling.

File: README.md (hunk 1 above)
Comment (wwelling) [reply README.md:30]: Are you sure we want to do this? I could be wrong, but it seemed fairly consistent that all okapi module docker images use port 8081 in the container.

File: descriptors/ModuleDescriptor-template.json (hunk 3 above)
Comment (wwelling) [reply descriptors/ModuleDescriptor-template.json:12]: Not true.

File: src/main/java/org/folio/rest/camunda/service/CamundaApiService.java (hunk 6 above)
Comment (kaladay) [reply src/main/java/org/folio/rest/camunda/service/CamundaApiService.java:57]: The exception is still being thrown.

Comment (wwelling) [reply src/main/java/org/folio/rest/camunda/service/CamundaApiService.java:57]: Okay, I will remove the exception from the error log.

File: src/test/resources/application.yaml [hunk 8]
```diff
@@ -91,7 +93,8 @@ spring:
     mode: TEXT
     suffix: .sql
 
-  jersey.application-path: rest
+  # this must be `camunda` to match module descriptor path patterns
+  jersey.application-path: camunda
```
Comment (wwelling) [inline src/test/resources/application.yaml:97]: Update [here](https://github.com/folio-org/mod-workflow/pull/117/files#diff-5f1399fae635ac57e696d503eb2135f7575b3a59a486e21613d3343f8f2a3243R50) when refactoring module descriptor to workflow-engine.

**Review by kaladay — APPROVED** [review]
Approving only to marshall things along.
